File: src/main/java/easy/ContainsNearbyDuplicate.java
package easy;

import java.util.HashMap;

public class ContainsNearbyDuplicate {
  public boolean solve(int[] nums, int k) {
    HashMap<Integer, Integer> seen = new HashMap<>();
    for (int i = 0; i < nums.length; i++) {
      if (seen.containsKey(nums[i])) {
        if (i - seen.get(nums[i]) <= k) return true;
      }
      seen.put(nums[i], i);
    }
    return false;
  }
}

File: src/main/php/Easy/ContainsNearbyDuplicate.php
<?php

declare(strict_types=1);

namespace Src\Easy;

class ContainsNearbyDuplicate
{
    /**
     * @param  int[]  $nums
     */
    public function solve(array $nums, int $k): bool
    {
        if (empty($nums)) {
            return false;
        }
        $result = [];
        foreach ($nums as $i => $num) {
            if (isset($result[$num]) && $i - $result[$num] <= $k) {
                return true;
            }
            $result[$num] = $i;
        }

        return false;
    }
}
